back/engines/commercial/insights/app/models/insights/view.rb

```diff
@@ -32,6 +27,15 @@ class View < ::ApplicationRecord
     )
 
     validates :name, presence: true, uniqueness: true
-    validates :scope, presence: true
+
+    accepts_nested_attributes_for :data_sources
+
+    def scope
+      data_sources.first.origin
+    end
+
+    def scope_id
+      data_sources.first.origin_id
+    end
```

Comment:
Those two methods are here as scaffolding and to make incremental changes easier. They will be removed at term.